cmd/serverless-init/log/log.go

```diff
@@ -54,30 +57,61 @@ func CreateConfig(origin string) *Config {
 }
 
 // SetupLogAgent creates the log agent and sets the base tags
-func SetupLogAgent(conf *Config, tags map[string]string, tagger tagger.Component, compression logscompression.Component) logsAgent.ServerlessLogsAgent {
+func SetupLogAgent(conf *Config, tags map[string]string, tagger tagger.Component, compression logscompression.Component, origin string) logsAgent.ServerlessLogsAgent {
 	logsAgent, _ := serverlessLogs.SetupLogAgent(conf.Channel, sourceName, conf.source, tagger, compression)
 
 	tagsArray := serverlessTag.MapToArray(tags)
 
-	addFileTailing(logsAgent, conf.source, tagsArray)
+	addFileTailing(logsAgent, conf.source, tagsArray, origin)
 
 	serverlessLogs.SetLogsTags(tagsArray)
 	return logsAgent
 }
 
-func addFileTailing(logsAgent logsAgent.ServerlessLogsAgent, source string, tags []string) {
-	if filePath, set := os.LookupEnv(envVarTailFilePath); set {
-		src := sources.NewLogSource("serverless-file-tail", &logConfig.LogsConfig{
+func addFileTailing(logsAgent logsAgent.ServerlessLogsAgent, source string, tags []string, origin string) {
+	// The Azure App Service log volume is shared across all instances. This leads to every instance tailing the same files.
+	// To avoid this, we want to add the azure instance ID to the filepath so each instance tails their respective system log files.
+	appServiceDefaultLoggingEnabled := origin == "appservice" && isInstanceTailingEnabled()
+
+	if appServiceDefaultLoggingEnabled {
```

Comment:
It seems like if instance tailing is disabled and `DD_SERVERLESS_LOG_PATH` is set to something other than `/home/LogFiles/*.log`, we'll still end up missing system logs?

Reply:
Yeah that's correct.